lib/matplotlib/backends/qt_editor/figure_plugin.py

```diff
@@ -0,0 +1,85 @@
+"""
+Plugin for drag and drop matplotlib.Figure in QtDesigner
+"""
+from matplotlib.backends.backend_qt5 import FigureCanvasQT
+from matplotlib.figure import Figure
+
+# Pyside and Pyside2 do not support the QtDesigner functionality that is
+# contained in both PyQt4 and PyQt5. This feature will not be supported with
+# those backends until this feature set is available in those libraries
+from matplotlib.backends.qt_compat import QtDesigner, QtGui
+
+
+class FigureDesignerPlugin(QtDesigner.QPyDesignerCustomWidgetPlugin):
+    """
+    QtDesigner Plugin for a matplotlib FigureCanvas
+
+    Notes
+    -----
+    In order to load this plugin, set the ``PYQTDESIGNERPATH`` environment
+    variable to the directory that contains this file.
+    """
+    def __init__(self):
+        QtDesigner.QPyDesignerCustomWidgetPlugin.__init__(self)
+        self.initialized = False
+
+    def initialize(self, core):
+        """Mark the QtDesigner plugin as initialized"""
+        if self.initialized:
+            return
+        self.initialized = True
+
+    def isInitialized(self):
+        """Whether the widget has been initialized"""
+        return self.initialized
+
+    def createWidget(self, parent):
+        """Create a FigureCanvasQT instance"""
+        # Create the Canvas with a new Figure
+        fig = FigureCanvasQT(Figure())
```

Comment:
Can you explain this a bit? I'm a bit surprised that you already add the figure to the canvas here. What if the user wants to use a different figure? (This may be a stupid question, as I've never worked with Designer.)

Reply:
I'm not sure of a way to pass in an existing `Figure` inside the `QtDesigner`. The idea is that the user creates the `.ui` file before any Python code is ever written. They then can access this figure (as shown above) by loading the `.ui` file. I'm not sure how to swap this order and have the Designer be aware of an arbitrary `Figure` created in user code.

Reply:
I am happy with this order of operations an the `.ui` figure now provides a fully functional `Figure` to work with.